### DISCUSSION TITLE
Questions on unsteady propeller optimization

### DISCUSSION THREAD

**Discussion by albisb** [mdolab/dafoam#484] [General]
Hi all,
I am working on adjoint optimization of a propeller for my thesis and I wanted to use DAFoam. For this, I have some questions:
- I saw in a reply mentioning URANS is not supported. However, I saw in the libraries that unsteady solvers such as pisoFoam or pimpleFoam are implemented in DAFoam. Could you please elaborate on that?
- I saw some tutorials and all of them are based on topology optimization. In my case I wanted to optimize some operational parameters of the propeller, which are not the shape of it. To do so, I wanted to obtain the adjoint gradients using DAFoam after the simulation is done and I would use them for gradient optimization, but without doing any shape optimization in between. Is it possible to do that?
- Finally, for the transient methods that are implemented, I would like to know if there are memory saving techniques already implemented in DAFoam, such as receding horizon. Otherwise I am fine trying to implement something myself.
Thanks in advance

**Comment by friedenhe** [comment]
1. No URANS is not supported. The current piso and pimple solvers are for flow simulations only. We are working on extending it to unsteady problems.
2. Yes, it is doable. But you need to implement new functions to compute the derivs wrt the propeller parameters. Please check the DAFoamworkshop 2022 slides. 
3. No...
